model/common/src/icon4py/model/common/orchestration/halo_exchange.py

```diff
@@ -0,0 +1,325 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# This file is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+import copy
+import site
+import sys
+from typing import Any, ClassVar, Optional, Sequence
+
+
+try:
+    import dace
+except ImportError as e:
+    raise ImportError("DaCe is required for this module") from e
+
+from dace import dtypes
+from dace.memlet import Memlet
+from dace.properties import CodeBlock
+from dace.sdfg.state import SDFGState
+from gt4py.next.ffront.fbuiltins import Dimension
+
+import icon4py.model.common as common
+from icon4py.model.common import dimension as dims
+
+
+ghex_ptr_names = (
```

Comment:
Follow standard PEP8 conventions (https://peps.python.org/pep-0008/#constants)
```suggestion
GHEX_PTR_NAMES: Final = (
```